datadog_checks_dev/datadog_checks/dev/plugin/tox.py

```diff
@@ -51,20 +53,37 @@ def tox_configure(config):
 def add_style_checker(config, sections, make_envconfig, reader):
     # testenv:style
     section = '{}{}'.format(tox.config.testenvprefix, STYLE_CHECK_ENV_NAME)
+
+    dependencies = [
+        'flake8',
+        'flake8-bugbear',
+        'flake8-logging-format',
+        'black',
+        'isort[pyproject]>=4.3.15',
+    ]
+
+    commands = [
+        'flake8 --config=../.flake8 --enable-extensions=G .',
+        'black --check --diff .',
+        'isort --check-only --diff --recursive .',
+    ]
+
+    if sections['testenv'].get(TYPES_FLAG, 'false').lower() == 'true':
+        # For command line options accepted by mypy, see: https://mypy.readthedocs.io/en/stable/command_line.html
+        # Defaults to type-checking the entire integration package.
+        mypy_args = sections['testenv'].get(MYPY_ARGS_OPTION, 'datadog_checks')
+
+        dependencies.append('mypy>=0.761')
+        commands.append('mypy --config-file=../mypy.ini {}'.format(mypy_args))
```

Comment:
Should we provide a path to mypy by default ? e.g. `./datadog_checks`

Reply:
The intent is not to, at least *not now*, because most of the time we will *want* to specify an explicit path of "here's the stuff that's ready for type checking — leave the rest alone".

Reply:
Oh wait, but then the line above where we default to `datadog_checks` is wrong! 🤦‍♂

Reply:
Still think that `./datadog_checks` can be a sensible default. 

But not a blocker and can be changed later.

Reply:
@AlexandreYang Maybe you were suggesting something else, i.e. allow writing:

```ini
[testenv]
dd_mypy_args = base/checks/base.py
```

Instead of…

```ini
[testenv]
dd_mypy_args = ./datadog_checks/base/checks/base.py
```

In practice the only code we'll type-check will be within the `./datadog_checks` folder, so maybe the magic makes sense…

The problem is that since it's just a list of arguments here, we can't easily automatically pre-pend `./datadog_checks` because we don't necessarily know which arguments are file paths.

One workaround would be split things up a bit more into options vs files, since the only positional arguments to `mypy` is the list of files to type check…

```ini
[testenv]
dd_check_types = true
dd_mypy_options = --py2 --check-untyped-defs --more-command-line-options
dd_mypy_files = just/a/list/of/files.py relative/to/datadog_checks
```

Is this what you had in mind @AlexandreYang?

Reply:
@AlexandreYang Didn't see your last comment. OK, so let's stick to explicit arguments *first*, and maybe if we discover patterns we can update the plugin…